lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -1992,26 +2007,141 @@ quitStakePool ctx (ApiT wid) body = do
 -------------------------------------------------------------------------------}
 
 createMigrationPlan
-    :: forall n s k. ()
-    => ApiLayer s k
-        -- ^ Source wallet context
+    :: forall ctx n s k.
+        ( ctx ~ ApiLayer s k
+        , Bounded (Index (AddressIndexDerivationType k) 'AddressK)
+        , HardDerivation k
+        , IsOwned s k
+        , Typeable n
+        , Typeable s
+        , WalletKey k
+        )
+    => ctx
     -> ApiT WalletId
         -- ^ Source wallet
     -> ApiWalletMigrationPlanPostData n
+        -- ^ Target addresses
     -> Handler (ApiWalletMigrationPlan n)
-createMigrationPlan _ctx _wid _postData = do
-    liftHandler $ throwE ErrTemporarilyDisabled
+createMigrationPlan ctx (ApiT wid) postData = do
+    (rewardWithdrawal, _) <- mkRewardAccountBuilder @_ @s @_ @n ctx wid Nothing
+    withWorkerCtx ctx wid liftE liftE $ \wrk -> liftHandler $ do
+        (wallet, _, _) <- withExceptT ErrCreateMigrationPlanNoSuchWallet $
+            W.readWallet wrk wid
+        plan <- W.createMigrationPlan wrk wid rewardWithdrawal
+        failWith ErrCreateMigrationPlanEmpty $ mkApiWalletMigrationPlan
+            (getState wallet)
+            (view #addresses postData)
+            (rewardWithdrawal)
+            (plan)
+
+mkApiWalletMigrationPlan
+    :: forall n s. IsOurs s Address
+    => s
+    -> NonEmpty (ApiT Address, Proxy n)
+    -> Withdrawal
+    -> MigrationPlan
+    -> Maybe (ApiWalletMigrationPlan n)
+mkApiWalletMigrationPlan s addresses rewardWithdrawal plan
+    | Just selections <- maybeSelections =
```

Comment:
```case maybeSelections of```

Reply:
Ah yes, the pattern guard is a bit unnecessary here!

(I think it might be a hangover from a previous version of the function.)

I've fixed this in a671d6eb69202f89285f87cad0c034df9a9bf7bf.